### PR TITLE
[COST-4716] Prevent removal of stale tags linked to a mapping.

### PR DESCRIPTION
**Pull request by myersCody**
## Jira Ticket

[COST-4716](https://issues.redhat.com/browse/COST-4716)

## Description

This change will prevent removing a stale enabled tag key if it is linked to a tag mapping. 

## Testing
testing instructions:
1. Delete a key that you just mapped from the tag summary:
```DELETE FROM reporting_awstags_summary where key='com_redhat_rhel_usage';```

2. Check that the key would of been returned with the new sql structure:
```
WITH cte_tag_mapping AS(
    SELECT distinct
        array_agg(child_id) as children_uuids,
        array_agg(parent_id) as parent_uuids
    FROM reporting_tagmapping
)
SELECT * FROM reporting_enabledtagkeys etk
CROSS JOIN cte_tag_mapping tag_map
WHERE NOT EXISTS (
    SELECT 1
    FROM reporting_awstags_summary AS ts
    WHERE ts.key = etk.key
    AND etk.provider_type = 'AWS'
)
AND etk.enabled = true
AND etk.provider_type = 'AWS'
AND NOT etk.uuid = ANY(tag_map.children_uuids || tag_map.parent_uuids);
```

3. Create a tag mapping with the key.
url: http://localhost:8000/api/cost-management/v1/settings/tags/mappings/child/add/
body:
```
{
    "parent": "f1df9dba-e8f8-43b6-8489-41c645474fbe",
    "children": ["1299f779-46eb-41da-b60e-c95e1946650b", "d1376436-2726-4dad-a3c3-e45a7b9e10df", "3c21b1fd-fb98-4dce-8999-a8be4996724b"]
}
```
You can find the uuids to use here:
```
http://localhost:8000/api/cost-management/v1/settings/tags/?limit=100
```
4. Rerun the sql:
```
WITH cte_tag_mapping AS(
    SELECT distinct
        array_agg(child_id) as children_uuids,
        array_agg(parent_id) as parent_uuids
    FROM reporting_tagmapping
)
SELECT * FROM reporting_enabledtagkeys etk
CROSS JOIN cte_tag_mapping tag_map
WHERE NOT EXISTS (
    SELECT 1
    FROM reporting_awstags_summary AS ts
    WHERE ts.key = etk.key
    AND etk.provider_type = 'AWS'
)
AND etk.enabled = true
AND etk.provider_type = 'AWS'
AND NOT etk.uuid = ANY(tag_map.children_uuids || tag_map.parent_uuids);
 uuid | key | enabled | provider_type | children_uuids | parent_uuids
------+-----+---------+---------------+----------------+--------------
(0 rows)
```
See that the enabled tag would not be deleted.

5. Delete another value from the tag summary tables for each provide. 
Then post to the following endpoint:
`http://127.0.0.1:5042/api/cost-management/v1/enabled_tags/`
```
{
    "schema": "org1234567",
    "action": "remove_stale",
    "provider_type": "aws"
}
```
Note that the provider_type here is all lower case unlike our other endpoints. This will run the stale removal SQL for you. You should see the key not associated with a tag mapping disappear, but the one that is stay. 

## Notes

...

### CODE REVIEW

File: koku/masu/database/sql/aws/remove_stale_enabled_tags.sql [hunk 1]
```diff
@@ -1,5 +1,12 @@
 -- Delete stale enabled keys
+WITH cte_tag_mapping AS(
+    SELECT distinct
```
Comment (cgoodfred) [inline koku/masu/database/sql/aws/remove_stale_enabled_tags.sql:3]: question: What is the `distinct` here doing, shouldn't the `array_agg` just turn all of the child and parent uuids into one record anyway?

Comment (myersCody) [reply koku/masu/database/sql/aws/remove_stale_enabled_tags.sql:3]: Awesome feedback @cgoodfred! Thank you! Let me push up a change.

**Review by cgoodfred — APPROVED** [review]
LGTM, great stuff

**Comment by samdoran** [comment]
Test failures are unrelated to changes in this PR.

**Pull request merged**